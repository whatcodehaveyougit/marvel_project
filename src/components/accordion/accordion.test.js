import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Accordion from './accordion.component';

it('renders correctly', () => {
  const tree = renderer
    .create(<Accordion title="title example" description="example description" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('displays the clicked element', () => {
  render(<Accordion title="title example" description="example description" />);
  const accordionSummary = screen.getByTestId('accordion-summary');

  // Clicking on the accordion summary should display the description
  fireEvent.click(accordionSummary);
  const descriptionElement = screen.getByText('example description');
  expect(descriptionElement).toBeInTheDocument();

  // Clicking again should hide the description
  fireEvent.click(accordionSummary);
  expect(descriptionElement).not.toBeInTheDocument();
});