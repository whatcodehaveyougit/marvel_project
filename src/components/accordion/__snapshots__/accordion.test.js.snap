// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion renders correctly 1`] = `
<DocumentFragment>
  <div
    class="accordion-character-container border border-gray-200 rounded my-4"
  >
    <div
      class="accordion-summary p-4 cursor-pointer"
      data-testid="accordion-summary"
    >
      <h4
        class="font-bold"
      >
        title example
      </h4>
    </div>
  </div>
</DocumentFragment>
`;
